MontyTRC89/TombEngine
# Volume system refactor

Solves #893 and implements activator queue for volumes, which allows infinite amount of objects to activate same volume at the same time. Solves issues with multiple NPCs not reacting to same volume (e.g. @GeckoKid-JW subway escalator case).

Additionally, this PR introduces script methods for volumes and also changes intersection method of volumes with player, so rough box is used instead precise one, which prevents false "oscillating" containment events from happening.

**Things to test: **

- Check if previous volume setups work as before.
- Check if multi-activator volume setups now work properly (including branching cases, when different activators should activate different node paths or script branches).
- Check if volume states are correctly saved / reloaded in savegames.
- Check if volume script functions work as intended.
